### PR TITLE
Update telegraf to 1.26.0 and increase the waittime for telegraf to 120

### PR DESCRIPTION
**Pull request by jatakiajanvi12**

### CODE REVIEW

File: build/common/installer/scripts/tomlparser-prom-agent-config.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@
 @promFbitMemBufLimit = 200
 
 @waittime_port_25226 = 45
-@waittime_port_25228 = 60
```
Comment (wanlonghenry) [inline build/common/installer/scripts/tomlparser-prom-agent-config.rb:17]: will 90s sufficient or it must be 120s

File: .trivyignore [hunk 2]
```diff
@@ -1,19 +1,14 @@
...
-CVE-2023-25173
-GHSA-4xgv-j62q-h3rj
-GHSA-hxp2-xqf3-v83h
+CVE-2023-28841
+CVE-2023-28842 
+CVE-2023-25809
 
-#telegraf HIGH
-CVE-2022-41723
+# telegraf HIGH
```
Comment (ganga1980) [inline .trivyignore:7]: what's the plan on fixing these telegraf high vulnerabilities?

File: .trivyignore [hunk 3]
```diff
@@ -1,19 +1,14 @@
...
 CVE-2023-27561
+CVE-2023-28840
+CVE-2023-28642
 
 # ruby HIGH
```
Comment (ganga1980) [inline .trivyignore:12]: Do we need to upgrade ruby version to address these ruby HIGH vulns?

File: .trivyignore [hunk 4]
```diff
@@ -1,19 +1,14 @@
...
-#telegraf HIGH
-CVE-2022-41723
+# telegraf HIGH
 CVE-2023-27561
+CVE-2023-28840
+CVE-2023-28642
 
 # ruby HIGH 
 CVE-2023-28756
 CVE-2023-28755
```
Comment (ganga1980) [inline .trivyignore:14]: Do we need to upgrade ruby version to address these ruby HIGH vulns?

File: kubernetes/linux/setup.sh [hunk 5]
```diff
@@ -57,7 +57,7 @@ sudo tdnf install jq-1.6-1.cm2 -y
 #used to setcaps for ruby process to read /proc/env
 sudo tdnf install libcap -y
 
-sudo tdnf install telegraf-1.25.2 -y
+sudo tdnf install telegraf-1.26.0 -y
```
Comment (ganga1980) [inline kubernetes/linux/setup.sh:60]: did we test this upgrade works as expected?

File: build/common/installer/scripts/tomlparser-prom-agent-config.rb [hunk 6]
```diff
@@ -14,7 +14,7 @@
...
+@waittime_port_25228 = 120
```
Comment (ganga1980) [inline build/common/installer/scripts/tomlparser-prom-agent-config.rb:17]: want to double check, this change doesnt impact the initial liveness probes, right?

File: kubernetes/linux/setup.sh (hunk 5 above)
Comment (jatakiajanvi12) [reply kubernetes/linux/setup.sh:60]: Yes have tested this...Working as before. Any other scenarios we would like to test?

File: build/common/installer/scripts/tomlparser-prom-agent-config.rb (hunk 6 above)
Comment (jatakiajanvi12) [reply build/common/installer/scripts/tomlparser-prom-agent-config.rb:17]: yes based on the understanding all the necessary livenessprobe executables have been started before this so shouldnt't affect

File: .trivyignore (hunk 4 above)
Comment (jatakiajanvi12) [reply .trivyignore:14]: No, working on getting the default gems upgraded.

File: .trivyignore (hunk 2 above)
Comment (jatakiajanvi12) [reply .trivyignore:7]: Working with Eric from Mariner team to figure this out

File: build/common/installer/scripts/tomlparser-prom-agent-config.rb (hunk 1 above)
Comment (jatakiajanvi12) [reply build/common/installer/scripts/tomlparser-prom-agent-config.rb:17]: So, may be 90 is sufficient. Adding a higher value so we can see what is the max needed from telemetry and then make a call to reduce it accordingly

File: kubernetes/linux/setup.sh (hunk 5 above)
Comment (ganga1980) [reply kubernetes/linux/setup.sh:60]: thanks, lets also look the perf usage and verify no perf regressions, if you havent verified already.

Comment (jatakiajanvi12) [reply kubernetes/linux/setup.sh:68]: I did but I added the ruby changes as well now so once that image is ready I will attach the screenshots

Comment (jatakiajanvi12) [reply kubernetes/linux/setup.sh:68]: Memory - workingset
image

Cpu usage
image

File: .trivyignore
Comment (pfrcks) [inline .trivyignore:1]: looks like there are merge conflicts?

File: kubernetes/linux/setup.sh [hunk 7]
```diff
@@ -29,6 +29,14 @@ rm -rf /usr/lib/ruby/gems/3.1.0/gems/openssl-3.0.1
 rm /usr/lib/ruby/gems/3.1.0/specifications/default/find-0.1.1.gemspec
 rm -rf /usr/lib/ruby/gems/3.1.0/gems/find-0.1.1
 
+# update the time and uri package to tackle the vulnerabilities in these gems
+gem update time --default
```
Comment (wanlonghenry) [inline kubernetes/linux/setup.sh:33]: is tehre any way to directly uninstall time and uri such as deleting the folder or use gem uninstall

File: kubernetes/linux/setup.sh (hunk 5 above)
Comment (pfrcks) [reply kubernetes/linux/setup.sh:68]: @jatakiajanvi12 not sure what these graphs mean, can you add legends ?

File: kubernetes/linux/setup.sh (hunk 7 above)
Comment (jatakiajanvi12) [reply kubernetes/linux/setup.sh:33]: No we are not deleting the gems, we are updating the gem and deleting the old version

File: kubernetes/linux/setup.sh (hunk 5 above)
Comment (jatakiajanvi12) [reply kubernetes/linux/setup.sh:68]: @pfrcks Updated the comment....Do you want me to add details in the graph?

File: kubernetes/linux/setup.sh [hunk 8]
```diff
@@ -29,6 +29,14 @@ rm -rf /usr/lib/ruby/gems/3.1.0/gems/openssl-3.0.1
 rm /usr/lib/ruby/gems/3.1.0/specifications/default/find-0.1.1.gemspec
 rm -rf /usr/lib/ruby/gems/3.1.0/gems/find-0.1.1
 
+# update the time and uri package to tackle the vulnerabilities in these gems
```
Comment (ganga1980) [inline kubernetes/linux/setup.sh:32]: @jatakiajanvi12 , I didnt see the change  for logging  Mariner version t in main.sh, will this be another PR?

Comment (jatakiajanvi12) [reply kubernetes/linux/setup.sh:32]: Yes let me add it in another PR...I havent tested that yet

File: kubernetes/linux/setup.sh (hunk 7 above)
Comment (wanlonghenry) [reply kubernetes/linux/setup.sh:33]: can we uninstall first then upgrade to save the mv?

Comment (jatakiajanvi12) [reply kubernetes/linux/setup.sh:33]: No we cant uninstall default gem...So, what we are doing is upgrade, get new version, move the older files so the new file is now default...Then gem uninstall the older version

**Review by ganga1980 — APPROVED** [review]

**Review by wanlonghenry — APPROVED** [review]

**Pull request merged**